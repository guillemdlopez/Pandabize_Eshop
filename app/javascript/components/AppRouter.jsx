import React from "react";
import { Route, BrowserRouter as Router, Switch } from "react-router-dom";
import BicycleForm from "./BicycleForm";
import BicycleGrid from "./BicycleGrid";
import BicycleShow from "./BicycleShow";
import Dashboard from "./Dashboard";

const AppRouter = () => {
  return (
    <div>
      <Router>
        <Switch>
          <Route exact path="/" component={BicycleGrid} />
          <Route exact path="/bicycle/:bicycleId" component={BicycleShow} />
          <Route
            exact
            path="/bicycle/:bicycleId/order"
            component={BicycleForm}
          />
          <Route exact path="/dashboard" component={Dashboard} />
        </Switch>
      </Router>
    </div>
  );
};

export default AppRouter;
