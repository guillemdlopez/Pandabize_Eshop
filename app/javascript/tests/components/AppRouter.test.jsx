import { mount } from "enzyme";
import React from "react";
import AppRouter from "../../components/AppRouter";
import { userContext } from "../../context/userContext";

describe("Testing the <AppRouter/> component", () => {
  const userInfo = {
    user: "Guillem",
    userId: 1,
  };
  test("should render it correctly", async () => {
    const wrapper = mount(
      <userContext.Provider value={userInfo}>
        <AppRouter />
      </userContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
