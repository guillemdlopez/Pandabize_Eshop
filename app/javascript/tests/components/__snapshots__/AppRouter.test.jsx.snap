// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <AppRouter/> component should render it correctly 1`] = `
<div>
  Array [
    <div
      className="banner"
    >
      <div
        className="container"
      >
        <h1
          className="banner-greeting"
        >
          Welcome back, 
          <strong>
            Guillem
          </strong>
        </h1>
      </div>
    </div>,
    <div
      className="container"
    >
      <div
        className="cards mb-5 mt-4 animate__animated animate__fadeIn"
      />
    </div>,
  ]
</div>
`;
